src/lib/prov/pkcs11/p11_mechanism.cpp

```diff
@@ -75,6 +75,11 @@ static std::map<std::string, RSA_SignMechanism> SignMechanisms =
    {
       { "Raw", RSA_SignMechanism(MechanismType::RsaX509) },
 
+      { "EMSA1(SHA-1)", RSA_SignMechanism(MechanismType::DsaSha1) },
```

Comment:
Note that these are ambiguous. We could come from either DSA or ECDSA Signature Schemes. Do you think it would be better not to allow them at all here?

Reply:
As we cannot handle this unambiguously it is probably better to not have them and fail instead.